packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -82,24 +98,39 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, {}>
             }];
             return (
                 <div className={className}>
-                    <div className="bp-table-truncated-value">{cellContent}</div>
+                    <div className="bp-table-truncated-value" ref={this.handleContentDivRef}>{cellContent}</div>
                     <Popover
                         className="bp-table-truncated-popover-target"
                         constraints={constraints}
                         content={popoverContent}
                         position={Position.BOTTOM}
-                        useSmartArrowPositioning={true}
+                        useSmartArrowPositioning
+                        useSmartPositioning
                     >
                         <span className="pt-icon-standard pt-icon-more"/>
                     </Popover>
                 </div>
             );
         } else {
             const className = classNames(this.props.className, "bp-table-truncated-text");
-            return <div className={className}>{cellContent}</div>;
+            return <div className={className} ref={this.handleContentDivRef}>{cellContent}</div>;
         }
     }
 
+    public componentDidMount() {
+        this.setTruncationState();
+    }
+
+    public componentDidUpdate() {
+        this.setTruncationState();
+    }
+
+    private handleContentDivRef = (ref: HTMLDivElement) => this.contentDiv = ref;
+
+    private isDetectingTruncation() {
```

Comment:
nit: not sure this function is necessary? anywhere it is used, it could instead be destructured from props